xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/MultiUserChatServiceImpl.java

```diff
@@ -1281,49 +1306,75 @@ public void initializeSettings() {
         }
     }
 
-    public void setLogMaxConversationBatchSize( int size )
+    @Override
+    public void setLogMaxBatchSize( int size )
     {
-        if ( this.logMaxConversationBatchSize == size ) {
+        if ( this.logMaxBatchSize == size ) {
             return;
         }
-        this.logMaxConversationBatchSize = size;
+        this.logMaxBatchSize = size;
 
-        archiver.setMaxWorkQueueSize( size );
+        if (archiver != null) {
+            archiver.setMaxWorkQueueSize(size);
+        }
         MUCPersistenceManager.setProperty( chatServiceName, "tasks.log.maxbatchsize", Integer.toString( size));
     }
 
-    public int getLogMaxConversationBatchSize()
+    @Override
+    public int getLogMaxBatchSize()
     {
-        return logMaxConversationBatchSize;
+        return logMaxBatchSize;
     }
 
+    @Override
     public void setLogMaxBatchInterval( Duration interval )
     {
         if ( this.logMaxBatchInterval.equals( interval ) ) {
             return;
         }
         this.logMaxBatchInterval = interval;
 
-        archiver.setMaxPurgeInterval( interval );
+        if (archiver != null) {
+            archiver.setMaxPurgeInterval(interval);
+        }
         MUCPersistenceManager.setProperty(chatServiceName, "tasks.log.maxbatchinterval", Long.toString( interval.toMillis() ) );
     }
 
+    @Override
     public Duration getLogMaxBatchInterval()
     {
         return logMaxBatchInterval;
     }
 
+    @Override
     public void setLogBatchGracePeriod( Duration period )
     {
         if ( this.logBatchGracePeriod.equals( period ) ) {
             return;
         }
 
         this.logBatchGracePeriod = period;
-        archiver.setGracePeriod( period );
+        if (archiver != null) {
+            archiver.setGracePeriod(period);
+        }
         MUCPersistenceManager.setProperty(chatServiceName, "tasks.log.batchgrace", Long.toString( period.toMillis() ) );
     }
 
+    @Override
+    public Duration getLogBatchGracePeriod()
+    {
+        return logBatchGracePeriod;
+    }
+
+    private Archiver getOrCreateConversationArchiver() {
```

Comment:
Do we need to worry about thread-safety here?

Reply:
I think we do. See new attempt in new commit.